tensorflow/python/keras/benchmark/bidirectional_lstm_benchmark_test.py

```diff
@@ -0,0 +1,139 @@
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Benchmark for Bidirectional LSTM on IMDB"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import timeit
+
+import numpy as np
+import six
+
+from tensorflow.python import keras
+from tensorflow.python.keras import layers
+from tensorflow.python.keras import callbacks
+from tensorflow.python.keras import preprocessing
+from tensorflow.python.keras.datasets import imdb
+from tensorflow.python.platform import benchmark
+from tensorflow.python.platform import test
+
+_NUM_EPOCHS = 2
+_MAX_FEATURE = 20000
+_MAX_LEN = 200
+
+
+class TimerCallBack(callbacks.Callback):
+  def __init__(self):
+    self.times = []
+    self.timer = timeit.default_timer
+    self.startup_time = timeit.default_timer()
+    self.recorded_startup = False
+
+  def on_epoch_begin(self, e, logs):
+    self.epoch_start_time = self.timer()
+
+  def on_epoch_end(self, e, logs):
+    self.times.append(self.timer() - self.epoch_start_time)
+
+  def on_batch_end(self, e, logs):
+    if not self.recorded_startup:
+      self.startup_time = self.timer() - self.startup_time
+      self.recorded_startup = True
+
+
+class BidirectLSTMBenchmark(
+    six.with_metaclass(benchmark.ParameterizedBenchmark, test.Benchmark)):
+  # Set different batch_size and iteration as needed.
+  _benchmark_parameters = [
+      ('bs_32', 32, 2), ('bs_64', 64, 2), ('bs_128', 128, 1),
+      ('bs_256', 256, 1), ('bs_512', 512, 3)]
+
+  def _measure_performance(self, model_fn, x_train, y_train, x_val, y_val,
```

Comment:
ditto: put this function into a util file, if it's commonly used in multiple benchmarks.

Reply:
For this, if we have to put into a file, we can create keras_benchmark, and we can inherit it when we have to build xxx_model_benchmark and call the same function name, such as _measure_performance(). Is that okay?

Reply:
> ditto: put this function into a util file, if it's commonly used in multiple benchmarks.
For this, if we have to put into a file, we can create keras_benchmark, and we can inherit it when we have to build xxx_model_benchmark and call the same function name, such as _measure_performance(). Is that okay?

Reply:
Yeah, makes sense to me! Thanks!